### ISSUE TITLE
Notes about Timeline appearance and changes in b.1.2.9.

### ISSUE THREAD

**Issue by oceanhahn** [aseprite/aseprite#1744]
- Having the relevant cels (current frame and layer intersections) in a new, darker colour on the Timeline is a great improvement, but losing the lighter-shade row and column backgrounds for irrelevant cels on selected layers and frames is a bit disconcerting. Is there a way to keep the unincluded cels on selected layers and frames at the second-dimmer colour? Perhaps this is intended and the current appearance is a bug.

 - Duplicating frames by CTRL-LMB-dragging them in the Timeline causes irrelevant cels in the selected (duplicate) frames to be lit using the dimmer of the two Timeline selection colours; deselects all but one layer. This only happens for frames; irrelevant cels in the remaining selected layer are not dimmed.
![1 2 9timeline](https://user-images.githubusercontent.com/11544843/40999411-c4dcf75e-68d8-11e8-84c0-7c17f6f2a811.gif)

 - CTRL- and SHIFT-LMB both select frames from the Timeline, but only SHIFT-LMB selects layers. CTRL-LMB probably ought to select layers as well.

 - It is not possible to select frames on the Timeline within a preexisting range. For instance, if frames 3 and 10 are selected, CTRL- or SHIFT-LMB on intervening frames (4~9) does not select those frames. It would be nice if the frame number under the cursor could be added or removed from the current selection when the cursor is not CTRL- or SHIFT-clicked, but not dragged (dragging duplicates frames as normal). It would be nice if we could deselect frames and layers this way as well.
(This issue exists in v1.2.8 as well.)

 - When multiple layers are selected, it would be handy if the various layer options (lock, show/hide, link new frames..) for all selected layers could be toggled with a single click, perhaps using the icon at the top of the layers list or by holding SHIFT or CTRL.

 - As a thought, when multiple selected frames or layers are moved, perhaps they ought to maintain their spacing _relative to one another,_ rather than piling up at the end of the new location? It is a bit hard to predict how frames and layers move this way. It may be difficult to work out how these frame moves will interact with frame tags.
(This was also an issue for frames in v.1.2.8.)

..using ASEprite beta.1.2.9 on Win10.

**Comment by oceanhahn** [comment]
> Not yet sure, I think Ctrl+click should start copying. So maybe shift+click should be the only way to increment selection (just like the rectangular marquee works on the canvas).

Are we making a distinction between CTRL-LMB-_Drag_ and CTRL-LMB here, or are they handled the same way?

> Actually I'm testing on Linux and I can click in frames inside the preexisting range with Shift+left mouse button. (Ctrl+click doesn't work, but Shift+click does work.)

Rechecked. SHIFT-LMB _does_ work. My bad; I probably got the whole CTRL-LMB-layer selection thing mixed up in my thoughts. x_x

Thanks for the speedy response!

**Comment by dacap** [comment]
> Are we making a distinction between CTRL-LMB-_Drag_ and CTRL-LMB here, or are they handled the same way?

:thinking: I guess that we can make a distintion, but it will complicate a little the code, so low priority at the moment.

> > Ctrl+click doesn't work, but Shift+click does work.
> Rechecked. SHIFT-LMB _does_ work.

:+1:

**Comment by dacap** [comment]
> losing the lighter-shade row and column backgrounds for irrelevant cels on selected layers and frames is a bit disconcerting

I didn't find a good way to fix this. With an extra color shade it's too confusing (too many shades). Maybe the timeline could have the same look as 1.2.8 when there isn't a selection of multiple frames, and be like 1.2.9 when there are multiple layers/frames.

**Comment by oceanhahn** [comment]
> I didn't find a good way to fix this. With an extra color shade it's too confusing (too many shades).

I suppose what I am imagining is something like this:
![image](https://user-images.githubusercontent.com/11544843/41044992-68d0d3ca-6975-11e8-90c2-3b0764deb162.png)
The red cel being the intersection of the current frame and active layer.

**Comment by dacap** [comment]
> I suppose what I am imagining is something like this

Yeah, I did something like that. There are too many shades, anyway I don't like the 1.2.9-beta1 look, the final result might be something similar to your screenshot.

**Comment by oceanhahn** [comment]
I have this other mockup, if you are interested in that look.
I do not like this version as much as the first I think, but here it is, for your consideration:
![image](https://user-images.githubusercontent.com/11544843/41055020-1bcecd20-698e-11e8-9611-ca2b6a24e3af.png)

**Comment by dacap** [comment]
Fixing this:

> Duplicating frames by CTRL-LMB-dragging them in the Timeline causes irrelevant cels in the new (duplicate) frames to be lit using the dimmer of the two Timeline selection colours

Bring the following issue too:

> when multiple selected frames or layers are moved, perhaps they ought to maintain their spacing relative to one another, rather than piling up at the end of the new location? It is a bit hard to predict how frames and layers move this way.

I guess that moving/copying multiple frames + layers will act like moving/copying a block.

**Comment by oceanhahn** [comment]
I have updated the top post in this thread to include a checklist, if that makes things easier for you.
I do not know what you define as "complete," so the boxes have been left unchecked.
If you do not want to curate it, I can do it as changes are moved to the beta or main branches. 😃

**Comment by dacap** [comment]
I've updated the [Steam beta branch](https://www.aseprite.org/beta/) with some changes about this.
